.cirrus.yml

```diff
@@ -25,10 +25,13 @@ final_docker_builder:
   only_if: $CIRRUS_BRANCH == "main"
   env:
     GITHUB_TOKEN: ENCRYPTED[98160948dd9e4613e0f7704958694e53fd7f021dd847938991e91c80e14ac5c91822aa9dff3fa34e52ec76f331beeeb5]
+  get_date_script: echo "DATE_TAG=$(date +%d%m%Y-%H%M%S)" >> $CIRRUS_ENV
   depends_on:
     - x86
     - arm
   login_script: echo $GITHUB_TOKEN | docker login ghcr.io -u davidchisnall --password-stdin
   manifest_script:
     - docker manifest create ghcr.io/cheriot-platform/devcontainer:latest --amend ghcr.io/cheriot-platform/devcontainer:x86_64-latest --amend ghcr.io/cheriot-platform/devcontainer:aarch64-latest
     - docker manifest push --purge ghcr.io/cheriot-platform/devcontainer:latest
+    - docker manifest create ghcr.io/cheriot-platform/devcontainer:$DATE_TAG --amend ghcr.io/cheriot-platform/devcontainer:x86_64-$DATE_TAG --amend ghcr.io/cheriot-platform/devcontainer:aarch64-$DATE_TAG
+    - docker manifest push --purge ghcr.io/cheriot-platform/devcontainer:$DATE_TAG
```

Comment:
Rather than relying on environment variables and coordination between the steps, it might be better for the x86 and Arm builds to just push the {arch}-latest tag and then this job push the combined one.  We probably don't actually need versioned architecture tags and, if I understand Docker manifests correctly (may not be true), the manifest that we're creating here refers to the hashes of the images identified by the tags, not to the tags themselves (if this were not the case, we wouldn't need to update the manifest each build, the existing one would transparently work as soon as we'd pushed the new versions of the arch ones).

As long as we don't run this job twice at the same time, it should be fine to just do the version for the multi-arch manifest and have it refer to whatever we successfully created for {arch}-latest.

Reply:
Ah - you might be right.   I'd had it in my head that the manifest kept the tag references, but that would make it all rather fragile.   Only having to create the date tag at the manifest level would make this a lot simpler.